### PR TITLE
changelog: update and regularize changelog entries

### PR DESCRIPTION
**Pull request by tychoish**
This adds a few missing change log entries and regularlizes the format/tense of the items.

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -32,13 +32,15 @@ Friendly reminder: We have a [bug bounty program](https://hackerone.com/tendermi
 - P2P Protocol
 
 - Go API
-  - [logging] \#6534 Removed the existing custom Tendermint logger backed by go-kit. The logging interface, `Logger`, remains.
+  - [node] \#6540 Reduce surface area of the `node` package by making most of the implementation details private. (@tychoish)
```
Comment (marbar3778) [inline CHANGELOG_PENDING.md:35]: for a few of these, they will require a few sentences in the upgrading.md. 

I know very few people read that 😞 but still worth it.

**Review by marbar3778 — COMMENTED** [review]
Looks good. Thanks for hopping on this. ❤️ 

i think there were some config changes as well, right?

**Comment by tychoish** [comment]
All config changes that I made that are in master are re in the change log. I think there are pending changes which have not yet been updated.

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (tychoish) [reply CHANGELOG_PENDING.md:34]: Yeah, I agree. What did you have in mind. The upgrading instructions seem to be compiled for all versions, don't seem to call out every Go API change, which means i think we make the 0.35 changes to the upgrading document in a more holistic manner. and we should probably think about this holistically (and also I don't think we're going to do a better job than the compiler).

Comment (marbar3778) [reply CHANGELOG_PENDING.md:34]: Breakdown of what is now internal and the new APIs compared to the previous ones. 

How did the node package, what do people need to know in order to upgrade. 

> (and also I don't think we're going to do a better job than the compiler).

huh? We do write api breaking changes in the changelog and upgrading doc even if the compiler helps. Not sure if you are suggesting passing it off to the compiler or not?

File: UPGRADING.md [hunk 2]
```diff
@@ -47,6 +47,17 @@ This guide provides instructions for upgrading to specific versions of Tendermin
...
 * CLI commands and flags are all now hyphen-case instead of snake_case.
   Make sure to adjust any scripts that calls a cli command with snake_casing
+
+### API Changes
+
+The p2p layer was reimplemented as part of the 0.35 release cycle, and
+all reactors were refactored. As part of that work these
+implementations moved into the `internal` package and are no longer
+considered part of the public Go API of tendermint. Accordingly, the
+space `node` package was changed to reduce access to tendermint
```
Comment (marbar3778) [inline UPGRADING.md:57]: What parts of the api should teams know about. If you look below we do mention how something's change, if we do not then its explained in the change log. I would prefer to keep the same structure as before, then pass it onto the compiler.

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (alexanderbez) [reply CHANGELOG_PENDING.md:34]: Yeah I think if in the upgrading doc, we just write a short section on what is now private and the 2-3 new node constructor/APIs, that should suffice.

File: UPGRADING.md [hunk 3]
```diff
@@ -47,6 +47,33 @@ This guide provides instructions for upgrading to specific versions of Tendermin
...
+- `evidence`
+
+Accordingly, the space `node` package was changed to reduce access to
+tendermint internals: applications that use tendermint as a library
+will need to change to accommodate these changes. Most notably:
+
+- The `Node` type has become internal, and all constructors return a
+  `service.Service` implementation.
+
+- The `node.DefaultNewNode` and `node.NewNode` constructorsare no
```
Comment (marbar3778) [inline UPGRADING.md:73]:
```suggestion
- The `node.DefaultNewNode` and `node.NewNode` constructors are no
```

**Review by marbar3778 — APPROVED** [review]
Noice. LGTM

**Pull request merged**